### PR TITLE
feat: Improve golden layout tab overflow drop down behaviour

### PR DESCRIPTION
**Pull request by dsmmcken**
- Header now has overflow hidden - which fixes an issue where the controls causing overflow of the viewport
- Since header has overflow, the dropdown menu is now moved to root while opened
- Add logic to flip menu to expand upward when there's no space below
- Add long name to test layout to check truncation behaviour
- Close menu when item is picked up based on a drag from the dropdown
- Resize dropdown when window is resized

Example of incorrect overflow:
Screenshot 2023-05-26 111108

### CODE REVIEW

File: packages/golden-layout/src/controls/Header.ts [hunk 1]
```diff
@@ -654,13 +660,14 @@ export default class Header extends EventEmitter {
...
   // enables synthetic keyboard navigation of the list
   _handleFilterKeydown(e: JQuery.TriggeredEvent) {
-    if (this.dropdownKeyIndex === -1) return;
-
     if (e.key === 'Escape') {
       this._hideAdditionalTabsDropdown();
       return;
     }
 
+    //escape should always close, even if list is empty
```
Comment (mofojed) [inline packages/golden-layout/src/controls/Header.ts:668]: Comment should have a space after `//` to be consistent with other comments.

That being said, I think comment here could be improved. This comment should should be before the escape key check above, then you could have a comment here saying something like `// a negative dropdownKeyIndex means the list is empty` (if that's the case, I'm inferring that from this comment saying something about the list being empty though I don't see why the list could have items in it but the dropdownKeyIndex isn't set for some reason).

File: packages/golden-layout/src/controls/Header.ts [hunk 2]
```diff
@@ -773,31 +780,60 @@ export default class Header extends EventEmitter {
     this.tabDropdownContainer.hide();
     this.isDropdownShown = false;
 
+    // put it back in the header dom
```
Comment (mofojed) [inline packages/golden-layout/src/controls/Header.ts:783]: _Why_ are we putting it back in the header DOM? Why don't we just create the container when we show the list and then remove it completely when it's hidden?

**Review by mofojed — CHANGES_REQUESTED** [review]

File: packages/golden-layout/src/controls/Header.ts (hunk 1 above)
Comment (dsmmcken) [reply packages/golden-layout/src/controls/Header.ts:668]: list can be empty when it's filtered to something non-matching.

File: packages/golden-layout/src/controls/Header.ts (hunk 2 above)
Comment (dsmmcken) [reply packages/golden-layout/src/controls/Header.ts]: Which would be easier if this was react and not jquery 🤷

**Review by mofojed — APPROVED** [review]

**Pull request merged**